eng/common/scripts/job-matrix/job-matrix-functions.ps1

```diff
@@ -190,11 +190,33 @@ function ParseFilter([string]$filter) {
     }
 }
 
-# Importing the JSON as PSCustomObject preserves key ordering,
-# whereas ConvertFrom-Json -AsHashtable does not
+function GetMatrixConfigFromFile([String] $config)
+{
+    [MatrixConfig]$config = try{
+        GetMatrixConfigFromJson($config)
```

Comment:
Should this take a file path or content? If we pass in the file path we can just check the file extension to choose.

Reply:
I think attempting to serialize is easier than inferring from extension name. People could name it yaml, yml, `.json.new`, etc. Right now the path is hardcoded for the yaml cases but I'd rather not restrict the file names as a json/yaml file can be valid despite the extension.